common-ui-compose/src/main/java/app/tivi/common/compose/AutoSizedCircularProgressIndicator.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app.tivi.common.compose
+
+import androidx.compose.Composable
+import androidx.ui.core.DensityAmbient
+import androidx.ui.core.Modifier
+import androidx.ui.core.WithConstraints
+import androidx.ui.graphics.Color
+import androidx.ui.layout.Column
+import androidx.ui.layout.size
+import androidx.ui.material.CircularProgressIndicator
+import androidx.ui.material.MaterialTheme
+import androidx.ui.material.Surface
+import androidx.ui.tooling.preview.Preview
+import androidx.ui.unit.dp
+import androidx.ui.unit.min
+
+@Composable
+fun AutoSizedCircularProgressIndicator(
+    modifier: Modifier = Modifier,
+    color: Color = MaterialTheme.colors.primary
+) {
+    WithConstraints(modifier) {
+        val diameter = with(DensityAmbient.current) {
+            min(constraints.maxWidth, constraints.maxHeight).toDp() - defaultPadding
```

Comment:
Hoist the padding?

Reply:
The padding is an internal constant in `CircularProgressIndicator` which I need to take into account to find the 'true' diameter.. Not much point hoisting it as it's not really a parameter imo.